src/erlcloud_glue.erl

```diff
@@ -0,0 +1,548 @@
+                glue_host         = Host,
+                glue_port         = Port,
+                glue_scheme       = Scheme,
+                retry             = fun erlcloud_retry:default_retry/1}.
+
+-spec configure(string(), string()) -> ok.
+configure(AccessKeyID, SecretAccessKey) ->
+    put(aws_config, new(AccessKeyID, SecretAccessKey)),
+    ok.
+
+-spec configure(string(), string(), string()) -> ok.
+configure(AccessKeyID, SecretAccessKey, Host) ->
+    put(aws_config, new(AccessKeyID, SecretAccessKey, Host)),
+    ok.
+
+-spec configure(string(), string(), string(), non_neg_integer()) -> ok.
+configure(AccessKeyID, SecretAccessKey, Host, Port) ->
+    put(aws_config, new(AccessKeyID, SecretAccessKey, Host, Port)),
+    ok.
+
+-spec configure(string(), string(), string(), non_neg_integer(), string()) -> ok.
+configure(AccessKeyID, SecretAccessKey, Host, Port, Scheme) ->
+    put(aws_config, new(AccessKeyID, SecretAccessKey, Host, Port, Scheme)),
+    ok.
+
+%% @doc
+%% Glue API:
+%% https://docs.aws.amazon.com/glue/latest/webapi/API_CreateJob.html
+%%
+-spec create_job(map()) -> ok | {error, any()}.
+create_job(JobInput) ->
+    create_job(JobInput, default_config()).
+
+-spec create_job(map(), aws_config()) -> ok | {error, any()}.
+create_job(JobInput, Config) ->
+    case request(Config, "CreateJob", JobInput) of
+        {ok, _} -> ok;
+        Error   -> Error
+    end.
+
+%% @doc
+%% Glue API:
+%% https://docs.aws.amazon.com/glue/latest/webapi/API_CreateTable.html
+%%
+%% `
+%% erlcloud_glue:create_table(<<"db-name">>,
+%%                            #{<<"Description">> => <<"some-desc">>,
+%%                              <<"Name">>        => <<"tbl-name">>,
+%%                              ...
+%%                              <<"TableType">>   => <<"some-type">>},
+%%                            <<"catalog-id">>).
+%% '
+%%
+-spec create_table(binary(), map()) -> ok | {error, any()}.
+create_table(DbName, TableInput) ->
+    create_table(DbName, TableInput, undefined, default_config()).
+
+-spec create_table(binary(),
+                   map(),
+                   binary() | aws_config()) ->
+    ok | {error, any()}.
+create_table(DbName, TableInput, CatalogId) when is_binary(CatalogId) ->
+    create_table(DbName, TableInput, CatalogId, default_config());
+create_table(DbName, TableInput, Config) when is_record(Config, aws_config) ->
+    create_table(DbName, TableInput, undefined, default_config()).
+
+-spec create_table(binary(), map(), binary() | undefined, aws_config()) ->
+    ok | {error, any()}.
+create_table(DbName, TableInput, CatalogId, Config) ->
+    Request0 = #{<<"DatabaseName">> => DbName,
+                 <<"TableInput">>   => TableInput},
+    Request1 = update_catalog_id(CatalogId, Request0),
+    case request(Config, "CreateTable", Request1) of
+        {ok, _} -> ok;
+        Error   -> Error
+    end.
+
+%% @doc
+%% Glue API:
+%% https://docs.aws.amazon.com/glue/latest/webapi/API_DeleteJob.html
+%%
```

Comment:
Any reason for not including a sample invocation here?

Reply:
i don't think it makes sense to document any obvious fun arguments and responses everywhere. We have spec for that, which looks descriptive in current case: jobname binary as input, ok as output:
```erlang
-spec delete_job(binary()) -> ok | {error, any()}.
delete_job(JobName) ->
    delete_job(JobName, default_config()).
```
The main problem with docs: they are forgotten to update very often
In other cases, where we have some not obvious arguments like `TableInput` as map, I've added docs

Reply:
👍